aten/src/ATen/native/Linear.cpp

```diff
@@ -541,9 +541,17 @@ Tensor einsum(c10::string_view equation, TensorList operands, at::OptionalIntArr
 
   // Sum out contraction dims
   if (perm_index - out_num_dim > 0) {
-    std::vector<int64_t> sum_dims(perm_index - out_num_dim);
-    std::iota(sum_dims.begin(), sum_dims.end(), out_num_dim);
-    ops[0] = ops[0].sum(sum_dims);
+    if (num_ops > 1) {
+      auto sizes = ops[0].sizes().vec();
+      for (auto dim = perm_index - 1; dim >= out_num_dim; --dim) {
+        sizes.erase(sizes.begin() + dim);
+      }
+      return ops[0].view(sizes);
+    } else {
+      std::vector<int64_t> sum_dims(perm_index - out_num_dim);
+      std::iota(sum_dims.begin(), sum_dims.end(), out_num_dim);
+      return ops[0].sum(sum_dims);
```

Comment:
`.sum(` -> `.sum_symint(`

Reply:
I think this has been eradicated with https://github.com/pytorch/pytorch/pull/83591

Reply:
Ho right, this side doesn't actually use sizes, it can remain int64_t !